contracts/proxies/GnosisSafeProxyFactory.sol

```diff
@@ -39,13 +24,23 @@ contract GnosisSafeProxyFactory {
     /// @param _singleton Address of singleton contract.
     /// @param initializer Payload for message call sent to new proxy contract.
     /// @param saltNonce Nonce that will be used to generate the salt to calculate the address of the new proxy contract.
+    /// @param includeChainIdInSalt If true, the chain id will be included in the salt to calculate the address of the new proxy contract.
     function deployProxyWithNonce(
         address _singleton,
         bytes memory initializer,
-        uint256 saltNonce
+        uint256 saltNonce,
+        bool includeChainIdInSalt
     ) internal returns (GnosisSafeProxy proxy) {
+        require(isContract(_singleton), "Singleton contract not deployed");
+
         // If the initializer changes the proxy address should change too. Hashing the initializer data is cheaper than just concatinating it
-        bytes32 salt = keccak256(abi.encodePacked(keccak256(initializer), saltNonce));
+        bytes32 salt;
+        if (includeChainIdInSalt) {
+            salt = keccak256(abi.encodePacked(keccak256(initializer), saltNonce, getChainId()));
```

Comment:
In my opinion it would be cleaner to calculate the `salt` in the corresponding methods (`createProxyWithNonce` and `createChainSpecificProxyWithNonce`) and pass the `salt` to this method.

This would avoid that we have a boolean input that has different execution paths here and it would make it very explicit in the corresponding methods what the difference is.

As part of this it probably would make sense to rename `deployProxyWithNonce` (e.g. just to `deployProxy`)